craft_parts/utils/partition_utils.py

```diff
@@ -51,8 +59,40 @@ def validate_partition_names(partitions: Optional[Sequence[str]]) -> None:
     if len(partitions) != len(set(partitions)):
         raise errors.FeatureError("Partitions must be unique.")
 
+    namespaces: Set[Tuple[str, str]] = set()
+    regular_partitions: Set[str] = set()
+
     for partition in partitions:
-        if not re.fullmatch("[a-z]+", partition):
+        # validate regular partitions
+        if re.fullmatch("[a-z]+", partition):
+            regular_partitions.add(partition)
+        # validate namespaced partitions
```

Comment:
Maybe add a check against duplicate entries? We can have strange side effects during processing if we allow a namespaced partition listed twice.